src/libraries/Hooks.sol

```diff
@@ -233,15 +229,18 @@ library Hooks {
     /// @notice calls beforeSwap hook if permissioned and validates return value
     function beforeSwap(IHooks self, PoolKey memory key, IPoolManager.SwapParams memory params, bytes calldata hookData)
         internal
-        returns (int256 amountToSwap, int128 hookDeltaSpecified)
+        returns (int256 amountToSwap, HookReturnDelta hookReturn)
     {
         amountToSwap = params.amountSpecified;
-        if (msg.sender == address(self)) return (amountToSwap, hookDeltaSpecified);
+        if (msg.sender == address(self)) return (amountToSwap, HookReturnDeltaLibrary.ZERO_DELTA);
+
         if (self.hasPermission(BEFORE_SWAP_FLAG)) {
-            hookDeltaSpecified = self.callHookWithReturnDelta(
+            hookReturn = self.callHookWithReturnDelta(
```

Comment:
Is there a way to know the unspecified delta amount in the beforeSwap hook? If i need to take a portion (say 0.1%) of the unspecified, is it possible to do it in the beforeSwap hook?